floris/tools/floris_interface.py

```diff
@@ -558,12 +558,20 @@ def check_wind_condition_for_viz(self, wd=None, ws=None):
         if len(ws) > 1 or len(ws) < 1:
             raise ValueError("Wind speed input must be of length 1 for visualization. Current length is {}.".format(len(ws)))
 
+    def check_calc_wake_run(self, func_name):
+
+        if not hasattr('self.floris.farm', 'turbine_type_map'):
+            raise AttributeError('Cant call function %s without first calling calculate_wake' % func_name)
+
     def get_turbine_powers(self) -> NDArrayFloat:
         """Calculates the power at each turbine in the windfarm.
 
         Returns:
             NDArrayFloat: [description]
         """
+
+        self.check_calc_wake_run('get_turbine_powers')
```

Comment:
Nice! We also need to add it to `get_farm_power`, and maybe some of the visualization functions?